src/coreclr/jit/importer.cpp

```diff
@@ -7531,6 +7531,63 @@ GenTree* Compiler::impImportStaticReadOnlyField(void* fldAddr, var_types lclTyp)
 {
     GenTree* op1 = nullptr;
 
+#if defined(DEBUG)
+    // If we're replaying under SuperPMI, we're going to read the data stored by SuperPMI and use it
+    // for optimization. Unfortunately, SuperPMI doesn't implement a guarantee on the alignment of
+    // this data, so for some platforms which don't allow unaligned access (e.g., Linux arm32),
+    // this can fault. We should fix SuperPMI to guarantee alignment, but that is a big change.
+    // Instead, simply fix up the data here for future use.
+
+    // This variable should be the largest size element, with the largest alignment requirement,
+    // and the native C++ compiler should guarantee sufficient alignment.
+    double aligned_data   = 0.0;
+    void*  p_aligned_data = &aligned_data;
+    if (info.compMethodSuperPMIIndex != -1)
+    {
+        switch (lclTyp)
+        {
+            case TYP_BOOL:
+            case TYP_BYTE:
+            case TYP_UBYTE:
+                static_assert_no_msg(sizeof(unsigned __int8) == sizeof(bool));
+                static_assert_no_msg(sizeof(unsigned __int8) == sizeof(signed char));
+                static_assert_no_msg(sizeof(unsigned __int8) == sizeof(unsigned char));
+                *(unsigned __int8*)p_aligned_data = *((unsigned __int8*)fldAddr);
+                break;
+
+            case TYP_SHORT:
+            case TYP_USHORT:
+                static_assert_no_msg(sizeof(unsigned __int16) == sizeof(short));
+                static_assert_no_msg(sizeof(unsigned __int16) == sizeof(unsigned short));
+                *(unsigned __int16*)p_aligned_data = GET_UNALIGNED_16(fldAddr);
```

Comment:
In these cases, you could check the current alignment of fldAddr and just use it if it meets the alignment requirement of 2,4 or 8

Reply:
@BruceForstall - Help me understand, but don't we want the workaround only in cases when `fldAddr` is unaligned?

Reply:
We only _need_ the workaround for unaligned `fldAddr`, but it doesn't hurt (except some miniscule perf -- in SuperPMI replay only) to do it. I went ahead and did it only when necessary.

Reply:
You mean you will move about chance only if `flAddr` is unaligned?

Reply:
Perfect. That's what I was expecting.

Reply:
FWIW I prefer what it was before. Actually, can't the entire thing just be:
```c++
    alignas(8) unsigned char aligned_data[8] = {};
    if (info.compMethodSuperPMIIndex != -1)
    {
        // Work around arm32 alignment bug in SuperPMI
        memcpy(aligned_data, fldAddr, min(sizeof(aligned_data), genTypeSize(lclTyp)));
        fldAddr = aligned_data;
    }
```
?

Anyway, I guess this workaround will disappear soon anyway, so no real need to do anything more.

Reply:
Yes, you're right that it could be smaller. Would I trust that `alignas` works for all compilers? Even so, could use `double` as I'm currently doing. And could do conditional, only-if-needed, as well. I think I'll leave it as is; there's not a lot of reason to optimize it.

Reply:
> Would I trust that alignas works for all compilers?

I believe alignas is part of C++11.

> I think I'll leave it as is; there's not a lot of reason to optimize it.

Agree, sounds fine to me.